### PR TITLE
Added json.md - Serialization part

### PR DESCRIPTION
**Pull request by Jai2305**
### Description
* Added documentation  json.md in guides folder for Json serialization with openSearch java Client.
* Added code samples to guide users to use the `toJsonString `method added [here ](https://github.com/opensearch-project/opensearch-java/pull/1064)and manual serialization with serialize method in `JsonpSerializable`
* Added a Working  Sample , created a new directory for json in /samples.

### Issues Resolved
Documentation for Json serialization with code samples
Work samples demonstrating logging with toJsonString method.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by Jai2305** [comment]
In response to @dblock's ask for documentation I have added json.md in guides folder with code samples demonstrating the usage of toJsonString.

File: guides/json.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+                  .aggregations(Collections.emptyMap())
+                  .terminateAfter(5L)
+                  .query(q -> q.match(t -> t.field("name").query(FieldValue.of("OpenSearch"))))
+  );
+}
+
+```
+### Using toJsonString method
+For classes implementing `PlainJsonSerializable`, which extends `JsonpSerializable`, a default `toJsonString` method is provided. 
+This method utilizes the default mapper and generator from the JsonpUtils class within the package to streamline the serialization process.
```
Comment (reta) [inline guides/json.md:28]:
```suggestion
This implementation uses `jakarta.json.spi.JsonProvider` SPI to discover the available JSON provider instance from the classpath and to create a new mapper. The `JsonpUtils` utility class streamlines this serialization process.
```

**Review by reta — DISMISSED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -16,6 +16,7 @@ This section is for maintaining a changelog for all breaking changes for the cli
 - Document HTTP/2 support ([#330](https://github.com/opensearch-project/opensearch-java/pull/330))
 - Add support for phase_took & search_pipeline request params ([#1036](https://github.com/opensearch-project/opensearch-java/pull/1036))
 - Add an interface PlainJsonSerializable inherit from JsonpSerializable with a default method streamlining serialization ([#1064](https://github.com/opensearch-project/opensearch-java/pull/1064))
+- Add a new file json.md in guides and SerializationBasics in samples/json ([#1083](https://github.com/opensearch-project/opensearch-java/pull/1083))
```
Comment (dblock) [inline CHANGELOG.md:19]: Nit: I would collapse the two changelog items and just say that we added request serialization to JSON.

File: guides/json.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
+- [Working with Json](#working-with-json)
+  - [Serialization](#serialization)
+    - [Using toJsonString method](#using-tojsonstring-method)
```
Comment (dblock) [inline guides/json.md:3]: Using `toJsonString` (quote, remove method)

File: guides/json.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    - [Manual serialization](#manually-serialize-by-providing-mapper-and-generator)
```
Comment (dblock) [inline guides/json.md:4]: Capitalize sections consistently, so "Manual Serialization"

File: guides/json.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+ 
+
+# Working with Json
+
+OpenSearch Java client seamlessly integrate with JSON providing serialization and deserialization capability.
```
Comment (dblock) [inline guides/json.md:9]: integrates

, providing

File: guides/json.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+
+## Serialization
+
+For demonstration let's consider an instance of `SearchRequest`.
+
+```java
+public static void main(final String[] args) {
```
Comment (dblock) [inline guides/json.md:16]: remove main, keep the body of it

File: guides/json.md (hunk 3 above)
Comment (dblock) [reply guides/json.md:3]: Update TOC to match body.

File: guides/json.md [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+      serialize(generator, mapper);
+    }
+    return writer.toString();
+  } catch (IOException ex) {
+    throw new UncheckedIOException(ex);
+  }
+}
+
+```
+
```
Comment (dblock) [inline guides/json.md:57]: Remove extra lines.

**Review by dblock — CHANGES_REQUESTED** [review]
Looks great, I have a bunch of nits, please.

Also link the new doc from USER_GUIDE

**Review by dblock — APPROVED** [review]

**Comment by dblock** [comment]
Merged, thank you!

**Pull request merged**